Formula/opentsdb.rb

```diff
@@ -0,0 +1,147 @@
+class HbaseLZORequirement < Requirement
+  fatal true
+
+  satisfy(:build_env => false) { Tab.for_name("hbase").with?("lzo") }
+
+  def message; <<-EOS.undent
+    hbase must not have disabled lzo compression to use it in opentsdb:
+      brew install hbase
+      not
+      brew install hbase --without-lzo
+    EOS
+  end
+end
+
+class Opentsdb < Formula
+  desc "Scalable, distributed Time Series Database."
+  homepage "http://opentsdb.net/"
+  url "https://github.com/OpenTSDB/opentsdb/releases/download/v2.2.0/opentsdb-2.2.0.tar.gz"
+  sha256 "5689d4d83ee21f1ce5892d064d6738bfa9fdef99f106f45d5c38eefb9476dfb5"
+
+  depends_on "hbase"
+  depends_on "lzo" => :recommended
+  depends_on HbaseLZORequirement if build.with?("lzo")
+  depends_on :java => "1.6+"
+  depends_on "gnuplot" => :optional
+
+  def install
+    # submitted to upstream: https://github.com/OpenTSDB/opentsdb/pull/711
+    # pulled to next branch: https://github.com/OpenTSDB/opentsdb/commit/5d0cfa9b4b6d8da86735efeea4856632581a7adb.patch
+    # doesn't apply cleanly on this release though
+    # mkdir_p is called from in a subdir of build so needs an extra ../ and there is no rule to create $(classes) and
+    # everything builds without specifying them as dependencies of the jar.
+    inreplace "Makefile.in" do |s|
+      s.sub!(/(\$\(jar\): manifest \.javac-stamp) \$\(classes\)/, '\1')
+      s.sub!(/(echo " \$\(mkdir_p\) '\$\$dstdir'"; )/, '\1../')
+    end
+
+    mkdir "build" do
+      system "../configure",
+             "--disable-silent-rules",
+             "--prefix=#{prefix}",
+             "--mandir=#{man}",
+             "--sysconfdir=#{etc}",
+             "--localstatedir=#{var}/opentsdb"
+      system "make"
+      bin.mkpath
+      system "make", "install-exec-am", "install-data-am"
+    end
+
+    env = {
+      :HBASE_HOME => Formula["hbase"].opt_libexec,
+      :COMPRESSION => (build.with?("lzo") ? "LZO" : "NONE"),
+    }
+    env = Language::Java.java_home_env.merge(env)
+    (bin/"create_table.sh").write_env_script opt_pkgshare/"tools/create_table.sh", env
```

Comment:
```diff
-    (bin/"create_table.sh").write_env_script opt_pkgshare/"tools/create_table.sh", env
+    create_table = pkgshare/"tools/create_table_with_env.sh"
+    create_table.write_env_script pkgshare/"tools/create_table.sh", env
+    create_table.chmod 0755
```

The problem here was that `write_env_script` doesn't bother to make the file executable (likely a bug), but instead relies on a post-install cleanup step to make files executable that are in locations like `bin`. Relocating it to `#{pkgshare}/tools/` caused it to end up in a location where that magic wouldn't kick in (and hence break the build).